lib/src/layer/polyline_layer.dart

```diff
@@ -56,29 +53,162 @@ class Polyline {
 @immutable
 class PolylineLayer extends StatelessWidget {
   final List<Polyline> polylines;
+  @Deprecated('has no effect anymore, polyline culling is enabled by default and controlled by margin')
   final bool polylineCulling;
 
+  /// extent outside of the viewport before culling polylines, set to null to
+  /// disable polyline culling
+  final double? polylineCullingMargin;
+
+  /// how much to simplify the polygons, in decimal degrees scaled to floored zoom
+  final double? simplificationTolerance;
+
+  /// use high quality simplification
```

Comment:
We should clearly define what high quality simplification is, and it's advantages/use-cases and disadvantages.